Mathlib/Data/Nat/FinMulAntidiagonal.lean

```diff
@@ -0,0 +1,246 @@
+/-
+Copyright (c) 2024 Arend Mellendijk. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arend Mellendijk
+-/
+
+import Mathlib.Algebra.Order.Antidiag.Pi
+import Mathlib.NumberTheory.ArithmeticFunction
+
+/-!
+# Sets of tuples with a fixed product
+
+This file defines the finite set of `d`-tuples of natural numbers with a fixed product `n` as
+`Nat.finMulAntidiagonal`.
+
+## Main Results
+* There are `d^(ω n)` ways to write `n` as a product of `d` natural numbers, when `n` is squarefree
+(`card_finMulAntidiagonal_of_squarefree`)
+-/
+
+open Finset
+open scoped BigOperators ArithmeticFunction
+namespace PNat
+
+/-- The set of divisors of a positive natural number.
+This is `Nat.divisorsAntidiagonal` without a special case for `n = 0`. -/
+def divisorsAntidiagonal (n : ℕ+) : Finset (ℕ+ × ℕ+) :=
```

Comment:
```suggestion
def divisorsAntidiag (n : ℕ+) : Finset (ℕ+ × ℕ+) :=
```
to match `piAntidiag` and avoid having super long names (if you want to rename the existing things called `antidiagonal`, I would be grateful!)